lib/pos_blockchain.go

```diff
@@ -12,23 +12,67 @@ import (
 	"github.com/pkg/errors"
 )
 
+// processHeaderPoS validates and stores an incoming block header to build
+// the PoS version of the header chain.
+//
+// The PoS header chain uses a simplified version of the Fast-HotStuff consensus
+// rules. It's used during syncing to build a chain of block headers with the
+// minimum set of validations needed to build a template of what the full PoS
+// blockchain will look like.
+//
+// The PoS header chain uses block integrity checks to perform block validations,
+// and to connect blocks based on their PrevBlockHash. It does not run the commit
+// rule. It does not fully validate QCs or block proposers, or perform any validations
+// that require on-chain state.
+//
+// processHeaderPoS algorithm:
+//  1. Validate that the block header is properly formed.
+//  2. Add the block header to the block index with status STORED.
```

Comment:
I think we'll want to store with the status `StatusHeaderValidated` if it passes validations and `StatusHeaderValidateFailed` if it fails validation. We're using the status `StatusBlockStored` to indicate that we've stored the full block including transactions.